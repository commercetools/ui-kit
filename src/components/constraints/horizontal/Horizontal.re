type constraint_ =
  | Scale
  | Xs
  | S
  | M
  | L
  | Xl;

module Styles = {
  open Css;

  let container = style([width(`percent(100.)), position(`relative)]);

  let style_of_constraint = c =>
    switch (c) {
    | Scale => style([width(`percent(100.))])
    | Xs => style([minWidth(`px(50))])
    | S => style([minWidth(`px(132))])
    | M => style([minWidth(`px(355))])
    | L => style([minWidth(`px(400))])
    | Xl => style([minWidth(`px(768))])
    };
};

let component = ReasonReact.statelessComponent(__MODULE__);

[@genType]
let make = (~constraint_=Scale, children) => {
  ...component,
  render: _self =>
    <div
      className=(
        Css.merge([
          Styles.container,
          Styles.style_of_constraint(constraint_),
        ])
      )>
      children
    </div>,
};

