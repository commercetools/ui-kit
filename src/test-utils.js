import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider, addLocaleData } from 'react-intl';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import en from 'react-intl/locale-data/en';
import de from 'react-intl/locale-data/de';
import es from 'react-intl/locale-data/es';
import messages from '../i18n/core.json';

addLocaleData(en);
addLocaleData(de);
addLocaleData(es);

const customRender = (
  node,
  {
    locale = 'en', // react-router
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    ...rtlOptions
  } = {}
) => ({
  ...render(
    <IntlProvider locale={locale} messages={messages}>
      <Router history={history}>{node}</Router>
    </IntlProvider>,
    rtlOptions
  ),
  // adding `history` to the returned utilities to allow us
  // to reference it in our tests (just try to avoid using
  // this to test implementation details).
  history,
});

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
