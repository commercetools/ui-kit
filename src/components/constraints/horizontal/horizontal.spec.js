import React from 'react';
import { render } from '../../../test-utils';
import Horizontal from './horizontal';

// The different constraints are tested in the visual regression tests.
// That tests also ensure this component accepts a "constraint" prop.

it('should render children', () => {
  const { getByTestId } = render(
    <Horizontal>
      <div data-testid="child" />
    </Horizontal>
  );
  expect(getByTestId('child')).toBeInTheDocument();
});
