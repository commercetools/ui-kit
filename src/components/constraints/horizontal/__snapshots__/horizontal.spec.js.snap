// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering by default should match snapshot 1`] = `
<div
  className="container constraintScale"
>
  <div />
</div>
`;

exports[`rendering with \`constraint\` should match snapshot 1`] = `
<div
  className="container constraintXl"
>
  <div />
</div>
`;
